apps/mobile-wallet/src/contexts/SendContext.tsx

```diff
@@ -102,52 +116,32 @@ export const SendContextProvider = ({
     [assetAmounts]
   )
 
-  const buildConsolidationTransactions = useCallback(async () => {
-    if (!address) return
-
-    try {
-      const data = await buildSweepTransactions(address, address.hash)
-      setUnsignedTxData(data)
-    } catch (e) {
-      showExceptionToast(e, t('Could not build transaction'))
-    }
-  }, [address, t])
-
   const sendTransaction = useCallback(
     async (onSendSuccess: () => void) => {
       if (!address || !toAddress) return
 
-      const { attoAlphAmount, tokens } = getTransactionAssetAmounts(assetAmounts)
-
       try {
-        for (const { unsignedTx } of unsignedTxData.unsignedTxs) {
-          const data = await signer.signAndSubmitUnsignedTx({ signerAddress: address.hash, unsignedTx })
-
-          dispatch(
-            transactionSent({
-              hash: data.txId,
-              fromAddress: address.hash,
-              toAddress: consolidationRequired ? address.hash : toAddress,
-              amount: attoAlphAmount,
-              tokens,
-              timestamp: new Date().getTime(),
-              status: 'sent',
-              type: 'transfer'
-            })
-          )
+        if (shouldSweep) {
+          const txParams = getSweepTxParams(address, toAddress)
+          await sendSweepTransactions(txParams)
+        } else if (chainedTxProps && chainedTxProps.length > 0 && groupedAddressWithEnoughAlphForGas) {
```

Comment:
Using a `shouldChainTxsForGasRefill` variable would make it clearer than using the generic `chainedTxProps` IMO.

Reply:
Alright! Done in 86f740a1f